bin/reth/src/node/mod.rs

```diff
@@ -234,6 +243,15 @@ impl<Ext: RethCliExt> NodeCommand<Ext> {
     pub async fn execute(mut self, ctx: CliContext) -> eyre::Result<()> {
         info!(target: "reth::cli", "reth {} starting", SHORT_VERSION);
 
+        // Runtime safety check for the `optimism` feature flag being enabled on a non-OP Stack
+        // chain.
+        #[cfg(feature = "optimism")]
+        if !self.chain.is_optimism() {
+            eyre::bail!(
+                "The `optimism` feature flag is enabled, but the chain is a non OP Stack chain."
+            )
+        }
```

Comment:
this only guards the node command, we have more commands that touch the DB,

this check should be a macro that generates a runtime error instead so we can add this to the main.rs so we can test correctness in CI via `reth --version`

Reply:
How can we have a macro with a runtime error that's dependent on the `self.chain.is_optimism()` method. I'm probably missing something but it doesn't seem like we can naively hoist this up to the main bin

Reply:
Moved the safety check up into the `Cli`'s `run` function, which should cover everything. Let me know if you'd prefer a different solution.

Reply:
Added separate binary in https://github.com/anton-rs/op-reth/pull/159